data/data/aws/vpc/vpc-public.tf

```diff
@@ -25,23 +25,23 @@ resource "aws_route" "igw_route" {
   gateway_id             = "${aws_internet_gateway.igw.id}"
 }
 
-resource "aws_subnet" "master_subnet" {
+resource "aws_subnet" "public_subnet" {
   count  = "${local.new_az_count}"
```

Comment:
Do we need to create a subnet for every availability zone when we are only going to use the first one? Are we keeping the public subnets around for the life of the cluster for other purposes where they are useful for other purpose?

Reply:
> Do we need to create a subnet for every availability zone when we are only going to use the first one?

We don't want to do this, but without #792 and #1121, fixing it is difficult.

Reply:
I don't see how those would be related here. The `public_subnet` is only used by the bootstrap machine now. It is not affected in any way by the master machines configuration.

Reply:
> The `public_subnet` is only used by the bootstrap machine now. It is not affected in any way by the master machines configuration.

Won't gateways and such also live in the public subnets?  We probably want one of those in each region that has master and/or compute nodes.

Reply:
OK. It isn't clear to me how the masters and workers would use those gateways since the machines are not part of the public subnet.